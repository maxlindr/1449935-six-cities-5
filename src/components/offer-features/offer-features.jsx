import React from 'react';
import PropTypes from 'prop-types';

function OfferFeatures(props) {
  return (
    <div className="property__inside">
      <h2 className="property__inside-title">What&apos;s inside</h2>
      <ul className="property__inside-list">
        {props.features.map((feature) => (
          <li key={feature} className="property__inside-item">
            {feature}
          </li>
        ))}
      </ul>
    </div>
  );
}

OfferFeatures.propTypes = {
  features: PropTypes.arrayOf(PropTypes.string).isRequired
};

export {OfferFeatures};
export default React.memo(OfferFeatures);
