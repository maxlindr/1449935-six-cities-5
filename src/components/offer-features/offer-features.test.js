import React from 'react';
import renderer from 'react-test-renderer';
import {OfferFeatures} from './offer-features';

const FEATURES = [`Feature 1`, `Feature 2`, `Feature 3`];

test(`OfferFeatures должен корректно отображаться`, () => {
  const component = renderer.create(
      <OfferFeatures features={FEATURES} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
