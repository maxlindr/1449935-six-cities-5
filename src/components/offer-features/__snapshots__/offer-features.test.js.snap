// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferFeatures должен корректно отображаться 1`] = `
<div
  className="property__inside"
>
  <h2
    className="property__inside-title"
  >
    What's inside
  </h2>
  <ul
    className="property__inside-list"
  >
    <li
      className="property__inside-item"
    >
      Feature 1
    </li>
    <li
      className="property__inside-item"
    >
      Feature 2
    </li>
    <li
      className="property__inside-item"
    >
      Feature 3
    </li>
  </ul>
</div>
`;
